jscomp/super_errors/super_misc.ml

```diff
@@ -0,0 +1,130 @@
+    let current_line = lines.(i - 1) in
+    (* disregard lines that are empty or are nothing but whitespace *)
+    if String.length (String.trim current_line) == 0 then ()
+    else
+      let current_line_leading_space_count = leading_space_count current_line in
+      match !columns_to_cut with
+      | None ->
+        columns_to_cut := Some current_line_leading_space_count
+      | Some n when n > current_line_leading_space_count ->
+        columns_to_cut := Some current_line_leading_space_count
+      | Some n -> ()
+  done;
+  let columns_to_cut = match !columns_to_cut with
+  | None -> 0
+  | Some n -> n
+  in
+  (* btw, these are unicode chars. They're not of length 1. Careful; we need to
+    explicitly tell Format to treat them as length 1 below *)
+  let separator = if columns_to_cut = 0 then "│" else "┆" in
+
+  fprintf ppf "@[<v 0>";
+  (* inclusive *)
+  for i = first_shown_line to last_shown_line do
+    let current_line = lines.(i - 1) in
+    let current_line_cut = current_line |> string_slice ~start:columns_to_cut in
+    let padded_line_number = pad (string_of_int i) max_line_number_number_of_digits in
+
+    fprintf ppf "@[<h 0>";
+
+    fprintf ppf "@[<h 0>";
+
+    (* this is where you insrt the vertical separator. Mark them as legnth 1 as explained above *)
+    if i < start_line || i > end_line then begin
+      (* normal, non-highlighted line *)
+      fprintf ppf "%s@{<separator> @<1>%s @}" padded_line_number separator
+    end else begin
+      (* highlighted row *)
+      fprintf ppf "@{<affected_line_number>%s@}@{<separator> @<1>%s @}" padded_line_number separator
+    end;
+
+    fprintf ppf "@]"; (* h *)
+
+    fprintf ppf "@[<hov 0>";
+
+    let current_line_strictly_between_start_and_end_line = i > start_line && i < end_line in
+
+    if current_line_strictly_between_start_and_end_line then fprintf ppf "@{<affected_line_content>";
+
+    let current_line_cut_length = String.length current_line_cut in
+    (* inclusive. To be consistent with using 1-indexed indices and count and i, j will be 1-indexed too *)
+    for j = 1 to current_line_cut_length do begin
+      let current_char = current_line_cut.[j - 1] in
+      if current_line_strictly_between_start_and_end_line then
+        fprintf ppf "%c@," current_char
+      else if i = start_line then begin
+        if j == (start_char - columns_to_cut) then fprintf ppf "@{<affected_line_content>";
+        fprintf ppf "%c@," current_char;
+        if j == (end_char - columns_to_cut) then fprintf ppf "@}"
+      end else if i = end_line then begin
+        if j == 1 then fprintf ppf "@{<affected_line_content>";
+        fprintf ppf "%c@," current_char;
+        if j == (end_char - columns_to_cut) then fprintf ppf "@}"
+      end else
+        (* normal, non-highlighted line *)
+        fprintf ppf "%c@," current_char
+    end
+    done;
+
+    if current_line_strictly_between_start_and_end_line then fprintf ppf "@}";
+
+    fprintf ppf "@]"; (* hov *)
```

Comment:
Did do that. Then lost track of boxes in the same function. I can refactor that later if needed